---
uint8 max_close
uint8 max_open